examples/node/pdf2png/pdf2png.js

```diff
@@ -70,41 +70,41 @@ const loadingTask = pdfjsLib.getDocument({
   cMapPacked: CMAP_PACKED,
   standardFontDataUrl: STANDARD_FONT_DATA_URL,
 });
-loadingTask.promise
-  .then(function (pdfDocument) {
-    console.log("# PDF document loaded.");
 
+(async function () {
+  try {
+    const pdfDocument = await loadingTask;
+    console.log("# PDF document loaded.");
     // Get the first page.
-    pdfDocument.getPage(1).then(function (page) {
-      // Render the page on a Node canvas with 100% scale.
-      const viewport = page.getViewport({ scale: 1.0 });
-      const canvasFactory = new NodeCanvasFactory();
-      const canvasAndContext = canvasFactory.create(
-        viewport.width,
-        viewport.height
-      );
-      const renderContext = {
-        canvasContext: canvasAndContext.context,
-        viewport,
-        canvasFactory,
-      };
-
-      const renderTask = page.render(renderContext);
-      renderTask.promise.then(function () {
-        // Convert the canvas to an image buffer.
-        const image = canvasAndContext.canvas.toBuffer();
-        fs.writeFile("output.png", image, function (error) {
-          if (error) {
-            console.error("Error: " + error);
-          } else {
-            console.log(
-              "Finished converting first page of PDF file to a PNG image."
-            );
-          }
-        });
+    const page = await pdfDocument.getPage(1);
+    // Render the page on a Node canvas with 100% scale.
+    const viewport = page.getViewport({ scale: 1.0 });
+    const canvasFactory = new NodeCanvasFactory();
+    const canvasAndContext = canvasFactory.create(
+      viewport.width,
+      viewport.height
+    );
+    const renderContext = {
+      canvasContext: canvasAndContext.context,
+      viewport,
+      canvasFactory,
+    };
+    const renderTask = page.render(renderContext);
+    (async function () {
+      await renderTask;
```

Comment:
Note that `renderTask` is *not* a Promise; please see https://github.com/mozilla/pdf.js/blob/f6d9d91965a97c80b6f5b335a27cdb4d36b1b9e2/src/display/api.js#L3171-L3203

Hence these lines should read:
```javascript

    const renderTask = page.render(renderContext);
    await renderTask.promise;
```
and there's no need to add *another* [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) here since you're already inside of an async function from above.